pkg/rpcclient/notary/actor.go

```diff
@@ -282,7 +284,10 @@ func (a *Actor) SendRequest(mainTx *transaction.Transaction, fbTx *transaction.T
 	if err != nil {
 		return mainHash, fbHash, vub, err
 	}
-	fbTx.Scripts[0].InvocationScript = append([]byte{byte(opcode.PUSHDATA1), keys.SignatureLen}, make([]byte, keys.SignatureLen)...) // Must be present.
+	// According to the Actor's constructor, Notary contract is always the last sender
+	// of the main transaction and always the first sender of the fallback transaction.
```

Comment:
You can't rely on that. `mainTx` can be anything.

Reply:
Notary subsystem itself doesn't rely on that and is able to work with both empty and dummy witnesses. However, documentation requires proper main transaction to have dummy notary witness. This requirement was added for proper contract-based witness verification cost calculation days ago when Notary fees where calculated on the client side without requests to RPC server. Now this requirement is obsolete and can be removed, I agree.

However, we have user's code relying on this dummy witness. We can safely update the neofs-node code to handle the new case of empty Notary invocation script, then OK, I'll just fix the neo-go's documentation.

Reply:
OK, I've fixed the doc.

Currently we have the following situation: empty Notary witness is allowed for main transaction and prohibited for fallback transaction. I don't quite like this situation, it leads to misunderstanding.